x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/filtering/FilteringValidation.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector.filtering;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+
+/**
+ * Represents the details of a validation process, including identifiers and descriptive messages.
+ * This class is used to encapsulate information about specific validation checks, where each validation
+ * is associated with a list of IDs and corresponding messages that detail the validation results.
+ */
+public class FilteringValidation implements Writeable, ToXContentObject {
+    private final List<String> ids;
```

Comment:
Are the messages related to the IDs? If so, shouldn't this be a List of tuples <id, message>?

Reply:
I doubt users would want to manually edit the basic/ advanced filtering rules by tinkering with raw JSON 😄 

Right now it's used by Kibana to manage the state of drafting and creation of new filtering rules for a connector. User will see this data when querying for the connector object. 

Not pasting an example directly here, but I'm linking the test from this PR that checks for data shape for connector filtering https://github.com/elastic/elasticsearch/pull/102592/files#diff-068f25a3a77ae7d85f0b826d1589f8cad2556673d509c845858e3a8bc3d803ab

Reply:
That could be changed for the user perspective so the two are aligned and remove the technical debt from the end users. But I'm ok with this - it's your domain and the API you're exposing, so it's your decision here 👍